common/src/primitives/height.rs

```diff
@@ -1,18 +1,113 @@
 use parity_scale_codec_derive::{Decode, Encode};
+use std::fmt;
+use std::ops::{Add, Sub};
+
+type HeightIntType = u64;
+type DistanceIntType = i64;
 
 #[derive(Debug, Copy, Clone, PartialOrd, Ord, PartialEq, Eq, Encode, Decode)]
-pub struct BlockHeight(u64);
+pub struct BlockHeight(HeightIntType);
+
+// Display should be defined for thiserr crate
+impl fmt::Display for BlockHeight {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        self.0.fmt(f)
+    }
+}
+
+impl From<BlockHeight> for HeightIntType {
+    fn from(block_height: BlockHeight) -> HeightIntType {
+        block_height.0
+    }
+}
+
+impl From<HeightIntType> for BlockHeight {
+    fn from(w: HeightIntType) -> BlockHeight {
+        BlockHeight(w)
+    }
+}
+
+impl Add<BlockDistance> for BlockHeight {
+    type Output = Option<BlockHeight>;
+
+    fn add(self, other: BlockDistance) -> Option<BlockHeight> {
+        let height: i64 = self.0.try_into().ok()?;
+        if height < 0 {
+            // we can't do arithmetic on this height anymore. Unless it's a bug, we won't face this in a million years
+            return None;
+        }
```

Comment:
Is the check still required? `self.0` is unsigned so it can't be negative and if it's too large to fit in `i64`, it should be already handled by `try_into`.

Reply:
@TheQuantumPhysicist ?

Reply:
No it's not required. It's just extra paranoia.